utils/translators/baidu_translation.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 The Pouch Robot Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translators
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// NewBaiduTranslator returns a translator of Baidu API
+func NewBaiduTranslator(options BaiduTranslatorOptions) Translator {
+	translator := baiduTranslator{
+		Appid: options.Appid,
+		Key:   options.Key,
+		From:  "auto",
+		To:    "en",
+	}
+	if options.From != "" {
+		translator.From = options.From
+	}
+	if options.To != "" {
+		translator.To = options.To
+	}
+	return &translator
+}
+
+type baiduTranslator struct {
```

Comment:
Could you add more descriptions about the fields?

Or drop an API link about this? @Frankzhaopku 

Since this would be much easier for us to check and review.

Reply:
Add description and API link here for review.